subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/PropertyValidationAccess.java

```diff
@@ -155,20 +155,19 @@ public String validate(boolean cacheable, PropertyMetadata metadata) {
         }
     }
 
-    private static class NestedTypeContext implements NestedBeanContext<TypeNode> {
+    private static class NestedTypeContext extends AbstractNestedBeanContext<TypeNode> {
 
-        private final PropertyMetadataStore metadataStore;
         private final Queue<TypeNode> queue;
 
         public NestedTypeContext(PropertyMetadataStore metadataStore, Queue<TypeNode> queue) {
-            this.metadataStore = metadataStore;
+            super(metadataStore);
             this.queue = queue;
         }
 
         @Override
         public TypeNode createNode(String propertyName, Object nested) {
             TypeToken<?> beanType = (TypeToken<?>) nested;
```

Comment:
Is there a reason not to generify the `Object` parameter here?

Reply:
The method is gone now.